master/internal/db/postgres_experiments.go

```diff
@@ -874,68 +873,75 @@ WHERE id = :id`
 	return db.namedExecOne(query, experiment)
 }
 
-// DeleteExperiment deletes an existing experiment.
-func (db *PgDB) DeleteExperiment(id int) error {
-	return db.withTransaction("delete experiment", func(tx *sqlx.Tx) error {
-		if _, err := tx.Exec(`
-DELETE FROM raw_steps
-WHERE trial_id IN (SELECT id FROM trials WHERE experiment_id = $1)
-`, id); err != nil {
-			return errors.Wrapf(err, "error deleting steps for experiment %v", id)
+// DeleteExperiments deletes one or more experiments.
+func (db *PgDB) DeleteExperiments(ctx context.Context, ids []int) error {
+	tx, err := Bun().BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		txErr := tx.Rollback()
+		if txErr != nil && txErr != sql.ErrTxDone {
+			log.WithError(txErr).Error("error rolling back transaction in DeleteExperiments")
 		}
+	}()
+	var delIDs []int32
+	if _, err = tx.NewDelete().Model(&delIDs).Table("raw_steps").
+		Where("trial_id IN (SELECT id FROM trials WHERE experiment_id IN (?))", bun.In(ids)).
+		Returning("id").
+		Exec(ctx); err != nil {
+		return errors.Wrapf(err, "error deleting steps for experiments %v", ids)
+	}
```

Comment:
ooof this is awkward. I'll add a ticket to track this: https://hpe-aiatscale.atlassian.net/browse/DET-9463